force-app/test/jest-mocks/lightning/navigation.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * For the original lightning/navigation mock that comes by default with
+ * @salesforce/lwc-jest, see:
+ * https://github.com/salesforce/lwc-jest/blob/master/src/lightning-mocks/navigation/navigation.js
+ */
+export const CurrentPageReference = jest.fn();
+
+let _pageReference, _replace;
+
+const Navigate = Symbol("Navigate");
+const GenerateUrl = Symbol("GenerateUrl");
+export const NavigationMixin = (Base) => {
+    return class extends Base {
+        [Navigate](pageReference, replace) {
+            _pageReference = pageReference;
+            _replace = replace;
+        }
+        [GenerateUrl](pageReference) {}
+    };
+};
+NavigationMixin.Navigate = Navigate;
+NavigationMixin.GenerateUrl = GenerateUrl;
+
+/*
+ * Tests will not have access to the internals of this mixin used by the
+ * component under test so we save a reference to what the Navigate method is
+ * called with and provide access to the params in the test via this function.
+ */
+export const getNavigateCalledWith = () => {
```

Comment:
I don't think we can get access to the Navigate function when it's used as a mixin on the component under test. This feels a bit wonky but it works and I don't think it's too terrible. Basically a very rudimentary `jest.fn()`.

Reply:
That's what I tried for a while but I ran into two issues:
1) Couldn't figure out the syntax to actually set `[Navigate]` to `jest.fn()`
2) Could not access the component under test's `[Navigate]` from the test case itself since it's a class mixin. `element[NavigationMixin.Navigate]`, for example, doesn't return anything.

Happy to try any specific code you have in mind.